### PR TITLE
refactor immutable dao code to align the new structure under pkg

### PR DESCRIPTION
**Pull request by wy65701436**
1, add manager
2, move model dao to /pkg/dao

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: src/pkg/immutabletag/dao/model/rule.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package models
+package model
```
Comment (ywk253100) [inline src/pkg/immutabletag/dao/model/rule.go]: Move this into immutabletag/model package if the manager and dao use the same one

File: src/pkg/immutabletag/model/rule.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	Kind string `json:"kind" valid:"Required;Match(doublestar)"`
+
+	// Decorated the selector
+	// for "doublestar" : "matching" and "excluding"
+	// for "label" : "with" and "without"
+	Decoration string `json:"decoration" valid:"Required"`
+
+	// Param for the selector
+	Pattern string `json:"pattern" valid:"Required"`
+}
```
Comment (stonezdj) [inline src/pkg/immutabletag/model/rule.go]: There is a similar Metadata struct in src/pkg/retention/policy/rule/models.go, should we reuse this struct?

File: src/pkg/immutabletag/manager.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func (drm *defaultRuleManager) UpdateImmutableRule(projectID int64, ir *model.Metadata) (int64, error) {
+	daoRule := &dao_model.ImmutableRule{}
+	data, _ := json.Marshal(ir)
+	daoRule.TagFilter = string(data)
+	return drm.dao.UpdateImmutableRule(projectID, daoRule)
+}
+
+func (drm *defaultRuleManager) EnableImmutableRule(id int64, enabled bool) (int64, error) {
```
Comment (stonezdj) [inline src/pkg/immutabletag/manager.go]: the name is confusing? If I want to disable any of rule, I still have to call EnableImmutableRule()

File: src/pkg/immutabletag/controller.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	// UpdateImmutableRule ...
+	UpdateImmutableRule(pid int64, m *model.Metadata) error
+
+	// ListImmutableRules ...
+	ListImmutableRules(pid int64) ([]model.Metadata, error)
+}
+
+// DefaultAPIController ...
+type DefaultAPIController struct {
+	manager Manager
```
Comment (stonezdj) [inline src/pkg/immutabletag/controller.go]: The controller just delegate the request for manager. do we need controller for all modules?

File: src/pkg/immutabletag/model/rule.go (hunk 2 above)
Comment (wy65701436) [reply src/pkg/immutabletag/model/rule.go]: discussed with SZ, we will not share any model between components under ./pkg..

File: src/pkg/immutabletag/dao/model/rule.go (hunk 1 above)
Comment (wy65701436) [reply src/pkg/immutabletag/dao/model/rule.go]: it's for dao only

**Pull request merged**